File: 08_Swift/SwiftProject/TestPods_OC/TestPods_OC/ViewController.m
//
//  ViewController.m
//  TestPods_OC
//
//  Created by 启业云03 on 2021/3/23.
//

#import "ViewController.h"

//#import <OnlyOCDemo/AllHeader.h>
//#import <OnlyOCDemo/FunctionB.h>
//
//#import <OnlySwiftDemo/OnlySwiftDemo-Swift.h>
//
//#import <OCAddSwiftDemo/FunctionD_OC.h>
//#import <OCAddSwiftDemo/OCAddSwiftDemo-Swift.h>

#import <Masonry.h>

//#import <QYCH5/QYCH5-Swift.h>
#import "QYCH5-Swift.h"
//@import QYCH5;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.

    self.view.backgroundColor = UIColor.orangeColor;
}

- (void)touchesBegan:(NSSet<UITouch *> *)touches withEvent:(UIEvent *)event {
    /*
    // 第一类：纯OC组件
    // [FunctionA printSomethingA:@"A打印"];
    [FunctionB printSomethingB:@"B打印"];
    
    // 第二类：纯Swift组件
    SwiftLibC *C = [[SwiftLibC alloc] init];
    [C show];
    
    // 第三类：OC组件包含Swift
    // OC组件中的OC方法
    [FunctionD_OC printSomething:@"你大爷！"];
    // OC组件中的OC方法再嵌套 Swift Pod 以及 Swift源码
    [FunctionD_OC useSwiftPod];
    [FunctionD_OC useSwiftSource];
    
    // OC组件中Swift源码
    SwiftLibD *D = [[SwiftLibD alloc] init];
    [D show:@"123"];
    // OC组件中Swift源码中调用OC源码以及Swift Pod
    [D show_OC_Source:@"789"];
    [D show_SwiftPod:@"10"];
     */
    
    ContainerWebViewVC *vc = [[ContainerWebViewVC alloc] init];
    [self presentViewController:vc animated:YES completion:nil];
}


@end

File: SwiftProject/TestPods_Swift/TestPods_Swift/ViewController.swift
//
//  ViewController.swift
//  TestPods_Swift
//
//  Created by 启业云03 on 2021/3/23.
//

import UIKit

import OnlyOCDemo

import OnlySwiftDemo

import OCAddSwiftDemo

import SwiftAddOCDemo

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        view.backgroundColor = .orange
    }
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        // 纯OC组件
        FunctionA.printSomethingA("你大爷！")
        FunctionB.printSomethingB("我二爷！")
        
        // 纯Swift组件
        OnlySwiftDemo.SwiftLibC().show()
        
        
        // OC组件包含Swift
        // OC组件中的OC方法
        FunctionD_OC.printSomething("123")
        FunctionD_OC.useSwiftPod()
        FunctionD_OC.useSwiftSource()
        
        // OC组件中Swift源码
        SwiftLibD().show("456")
        SwiftLibD().show_SwiftPod("789")
        SwiftLibD().show_OC_Source("10")
        
        
        // Swift组件包含OC
        FunctionF_OC.printSomethingF("777")
        
        SwiftLibE().show("1")
        SwiftLibE().show_OC_Source("2")
        SwiftLibE().show_OC_Source("3")
    }

}

